src/tlapm_args.ml

```diff
@@ -279,10 +366,44 @@ let init () =
     Array.iter (fun s -> Printf.printf " %s" (quote_if_needed s)) Sys.argv;
     Printf.printf "\n\n%!"
   end;
-  if !use_stdin && (List.length !mods) <> 1 then begin
-    Arg.usage opts
+  if !use_stdin && (List.length mods) <> 1 then begin
+    Printf.eprintf
       "Exactly 1 module has to be specified if TLAPM is invoked with\
-       the --stdin option." ;
+       the --stdin option.\n%!";
     exit 2
   end;
-  !mods
+  mods
+
+let term =
+  Term.(const main $ version_t $ verbose_t $ where_t $ config_t $ summary_t $ timing_t
+        $ search_dir_t $ keep_going_t $ suppress_all_t $ check_t $ threads_t $ method_t
+        $ solver_t $ smt_logic_t $ fast_isabelle_t $ stretch_t $ noflatten_t $ nonormal_t
+        $ debug_t $ toolbox_t $ toolbox_vsn_t $ line_t $ wait_t $ stdin_t $ prefer_stdlib_t
+        $ noproving_t $ printallobs_t $ safefp_t $ nofp_t $ nofpl_t $ cleanfp_t
+        $ erasefp_t $ printfp_t $ usefp_t $ fpp_t $ cache_dir_t $ files_t)
+
+let cmd =
+  let info = Cmd.info "tlapm" in
+  Cmd.v info term
+
+let no_comma s =
+  match String.index s ',' with
+  | exception Not_found -> true
+  | _ -> false
+
+(* Preprocess command line before feeding it to Cmdliner, to preserve
+   backward compatibility. In particular, Cmdliner does not support
+   options with direct multiple arguments. *)
```

Comment:
Could you add an example of the specific types of text that are being rewritten?